strax/processing/general.py

```diff
@@ -364,3 +364,69 @@ def _touching_windows(thing_start, thing_end,
         result[i] = left_i, right_i
 
     return result
+
+@export
+def abs_time_to_prev_next_interval(things, intervals):
+    """Function which determines the time difference of things to
+    previous and next interval, e.g., events to veto intervals. Assumes
+    that things do not overlap.
+    
+    :param things: Numpy structured array containing strax time fields
+    :param intervals: Numpy structured array containing time fields
+    :returns: Two integer arrays with the time difference to the 
+        previous and next intervals. 
+    """
+    _things_do_not_overlap = (strax.endtime(things)[:-1] - things['time'][1:]) <= 0
+    assert np.all(_things_do_not_overlap), 'Things overlap!'
+
+    times_to_prev = np.ones(len(things), dtype=np.int64) * -1
+    times_to_next = np.ones(len(things), dtype=np.int64) * -1
+    
+    _empty_events_or_intervals = ((len(things) == 0) 
+                                  or (len(intervals)== 0))
+    if _empty_events_or_intervals:
+        return times_to_prev, times_to_next
+        
+    _abs_time_to_prev_next(things, intervals, times_to_prev, times_to_next)
+    
+    return times_to_prev, times_to_next
+
+@numba.njit
+def _abs_time_to_prev_next(things, intervals, times_to_prev, times_to_next):
+    veto_intervals_seen = 0
+    last_thing_end = 0
+    for thing_ind, thing_i in enumerate(things):
+        current_event_time = thing_i['time']
+        current_event_endtime = strax.endtime(thing_i)
+        
+        # Exploit the fact that events cannot overlap...
+        # Loop over veto intervals until:
+        #   - current veto interval endtime starts overlapping with 
+        #    current event. This is the interval from which we need to 
+        #    start looping for the next event.
+        #   - current veto_interval time starts to be larger than current 
+        #    event time. Only then we can be sure we have computed the 
+        #    shortest time delay as endtime is not sorted...
+        for veto_interval in intervals[veto_intervals_seen:]:
+            _interval_start_after_thing = veto_interval['time'] >= current_event_time
+            if _interval_start_after_thing:             
+                break
+            
+            # Always update endtime until it becomes negative:
+            dt = current_event_time - strax.endtime(veto_interval)
+            _ends_befor_event = dt >= 0
+            if _ends_befor_event:
+                times_to_prev[thing_ind] = dt
+                veto_intervals_seen += 1
+        
+        # Now check if current veto is still within event or already after it:
+        for veto_interval in intervals[veto_intervals_seen:]:
+            _current_interval_before_thing_ends = veto_interval['time'] < current_event_endtime
+            if _current_interval_before_thing_ends:
+                continue
+
+            # Now current event is after event so store time:
```

Comment:
Hey, @WenzDaniel . I guess here you mean 'Now current veto is after event so store time'?

Reply:
Thanks for spotting yes.